api/agent/drivers/docker/docker.go

```diff
@@ -102,13 +145,36 @@ func NewDocker(conf drivers.Config) *DockerDriver {
 		}
 	}
 
+	imagesBeforeLoad, err := driver.docker.ListImages(context.Background())
+
 	if conf.DockerLoadFile != "" {
 		err = loadDockerImages(driver, conf.DockerLoadFile)
 		if err != nil {
 			logrus.WithError(err).Fatalf("cannot load docker images in %s", conf.DockerLoadFile)
 		}
 	}
 
+	if conf.MaxImageCacheSize != 0 {
+		driver.imageCache = NewCache(int64(conf.MaxImageCacheSize))
+
+		go func() {
+			images, err := driver.docker.ListImages(context.Background())
+			if err != nil {
+				logrus.WithError(err).Fatalf("cannot list docker images %s", err)
+			}
+			for _, i := range images {
+				if Contains(imagesBeforeLoad, i) {
```

Comment:
is it possible to have `loadDockerImages()` return the id of the image that it loaded?  If so, would it make sense to simplify this check to just lock the loaded docker image, or are we expecting to have more than one here?